x/mongo/driver/topology/server.go

```diff
@@ -285,10 +288,10 @@ func (s *Server) Disconnect(ctx context.Context) error {
 	close(s.done)
 	s.cancelCheck()
 
-	s.rttMonitor.disconnect()
 	s.pool.close(ctx)
 
 	s.closewg.Wait()
+	s.rttMonitor.disconnect()
```

Comment:
It does not cause any actual impact whether or not this line is moved. I put it here as a "fail-safe" that the disconnecting happens after the `update()` exits, so `rttMonitor.connect()` is more likely to be called beforehand.